Issues found when building Chakra with MSVC + /permissive-

Building Chakra with MSVC + /permissive- failed with error C4596. Similar issue with https://github.com/Microsoft/ChakraCore/issues/2862

Here is issue:
https://github.com/Microsoft/ChakraCore/blob/master/lib/Runtime/Base/ThreadContext.h#L1784
https://github.com/Microsoft/ChakraCore/blob/master/lib/Runtime/Base/ThreadContext.h#L1793

Failures:
  d:\chakra\src\lib\runtime\Base/ThreadContext.h(1784): error C4596: '{ctor}': illegal qualified name in member declaration [D:\Chakra\src\lib\Parser\Chakra.Parser.vcxproj]
         d:\chakra\src\lib\runtime\Base/ThreadContext.h(1793): error C4596: '{dtor}': illegal qualified name in member declaration [D:\Chakra\src\lib\Parser\Chakra.Parser.vcxproj]

The member function declaration above is inside the class ‘AutoDisableInterrupt’ and the prefix 'AutoDisableInterrupt::' before the member function name isn’t allowed. The fix is to remove it.

--- a/lib/Runtime/Base/ThreadContext.h
+++ b/lib/Runtime/Base/ThreadContext.h
@@ -1785,7 +1785,7 @@ extern void(*InitializeAdditionalProperties)(ThreadContext *threadContext);
 class AutoDisableInterrupt
 {
 public:
-    AutoDisableInterrupt::AutoDisableInterrupt(ThreadContext *threadContext, bool explicitCompletion = true)
+    AutoDisableInterrupt(ThreadContext *threadContext, bool explicitCompletion = true)
         : m_operationCompleted(false), m_interruptDisableState(false), m_threadContext(threadContext), m_explicitCompletion(explicitCompletion)
     {
         if (m_threadContext->HasInterruptPoller())
@@ -1794,7 +1794,7 @@ public:
             m_threadContext->GetInterruptPoller()->SetDisabled(true);
         }
     }
-    AutoDisableInterrupt::~AutoDisableInterrupt()
+    ~AutoDisableInterrupt()
     {
         if (m_threadContext->HasInterruptPoller())
         {

